### ISSUE TITLE
Passage Mode is not working as expected

### ISSUE THREAD

**Issue by SDS712** [ttlock/ttlock_flutter#5]
Passage Mode is not getting set for the input parameters `startDate` and `endDate` provided

Attached below is the code from the example app with changed `startDate` and `endDate` parameters for testing purposes.
The passage mode is active even at **20:23 IST** with the below config

```
      case Command.addPassageMode:
        TTLock.addPassageMode(
            TTPassageModeType.weekly, [4], null, 0, 60, lockData, () {
          _showSuccessAndDismiss("Success");
        }, (errorCode, errorMsg) {
          _showErrorAndDismiss(errorCode, errorMsg);
        });
        break;

```

**Comment by ttlock** [comment]
```
case Command.addPassageMode:
       //startDate = 0, endDate = 60.  The passage mode will active at 0:00 am - 1:00 am
        TTLock.addPassageMode(
            TTPassageModeType.weekly, [4], null, 0, 60, lockData, () {
          _showSuccessAndDismiss("Success");
        }, (errorCode, errorMsg) {
          _showErrorAndDismiss(errorCode, errorMsg);
        });
        break;
```
The passage mode is active at the wrong time because the lock time is inaccurate.
You can query and set the time through the SDK interface.

Get
```
case Command.getLockTime:
TTLock.getLockTime(lockData, (lockTimestamp) {
          _showSuccessAndDismiss("Time: $lockTimestamp ");
        }, (errorCode, errorMsg) {
          _showErrorAndDismiss(errorCode, errorMsg);
        });
```

Set
```
case Command.setLockTime:
        int timestamp = DateTime.now().millisecondsSinceEpoch;
        TTLock.setLockTime(timestamp, lockData, () {
          _showSuccessAndDismiss("Success");
        }, (errorCode, errorMsg) {
          _showErrorAndDismiss(errorCode, errorMsg);
        });
        break;

```

**Comment by ttlock** [comment]
```
case Command.addPassageMode:
       //startDate = 0, endDate = 60.  The passage mode will active at 0:00 am - 1:00 am
        TTLock.addPassageMode(
            TTPassageModeType.weekly, [4], null, 0, 60, lockData, () {
          _showSuccessAndDismiss("Success");
        }, (errorCode, errorMsg) {
          _showErrorAndDismiss(errorCode, errorMsg);
        });
        break;
```
The passage mode is active at the wrong time because the lock time is inaccurate.
You can query and set the time via the SDK interface.


```
case Command.getLockTime:
TTLock.getLockTime(lockData, (lockTimestamp) {
          _showSuccessAndDismiss("Time: $lockTimestamp ");
        }, (errorCode, errorMsg) {
          _showErrorAndDismiss(errorCode, errorMsg);
        });
```


```
case Command.setLockTime:
        int timestamp = DateTime.now().millisecondsSinceEpoch;
        TTLock.setLockTime(timestamp, lockData, () {
          _showSuccessAndDismiss("Success");
        }, (errorCode, errorMsg) {
          _showErrorAndDismiss(errorCode, errorMsg);
        });
        break;

```

**Comment by SDS712** [comment]
Hello @ttlock team,

It gave same results even after calling `TTLock.setLockTime` before `TTLock.addPassageMode` as you've recommended above.

Also I observed the passage mode was active for fullday even with following config:

```
case Command.addPassageMode:
        TTLock.addPassageMode(
            TTPassageModeType.weekly, [4], null, 0, 60, lockData, () {
          _showSuccessAndDismiss("Success");
        }, (errorCode, errorMsg) {
          _showErrorAndDismiss(errorCode, errorMsg);
        });
        break;
```

**Comment by ttlock** [comment]
When you set the new passage mode, you need clear the previous settings using clearAllPassageModes api, Or the previous settings will also work.

**Comment by SDS712** [comment]
> When you set the new passage mode, you need clear the previous settings using clearAllPassageModes api, Or the previous settings will also work.

Still same issue. Here's what I'm doing.

First, I'm setting the Lock Time using

```
case Command.setLockTime:
        int timestamp = DateTime.now().millisecondsSinceEpoch;
        TTLock.setLockTime(timestamp, lockData, () {
          _showSuccessAndDismiss("Success");
        }, (errorCode, errorMsg) {
          _showErrorAndDismiss(errorCode, errorMsg);
        });
```

Then, I'm clearing all existing passage modes (if any) using

```
case Command.clearAllPassageModes:
        TTLock.clearAllPassageModes(lockData, () {
          _showSuccessAndDismiss("Success");
        }, (errorCode, errorMsg) {
          _showErrorAndDismiss(errorCode, errorMsg);
        });
```
And then, I'm adding passage mode with following config
```
case Command.addPassageMode:
        TTLock.addPassageMode(
            TTPassageModeType.weekly, [6], null, 0, 60, lockData, () {
          _showSuccessAndDismiss("Success");
        }, (errorCode, errorMsg) {
          _showErrorAndDismiss(errorCode, errorMsg);
        });
```
And It remains active for full day

This issue we've been facing is using the sdk
It is working perfectly fine through gateway using the 
**[Config the passage mode of a lock](https://open.ttlock.com/doc/api/v3/lock/configPassageMode) - `https://api.ttlock.com/v3/lock/configPassageMode`** API. 

Though in the API there is no provision made to input `monthDays `as it is made available in the sdk.

**Comment by ttlock** [comment]
Sorry for the api bug, we will fix the bug in next version.You need wait a few days.

**Comment by SDS712** [comment]
fixed as part of v0.0.4
